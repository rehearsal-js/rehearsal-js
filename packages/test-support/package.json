{
  "name": "@rehearsal/test-support",
  "version": "3.0.1-beta",
  "description": "Test support utilities for rehearsal",
  "license": "BSD-2-Clause",
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist/src/index.d.ts",
      "import": "./dist/src/index.js"
    },
    "./*": "./*"
  },
  "main": "dist/src/index.js",
  "types": "dist/src/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "tsc -b",
    "fixtures:clean": "rm -rf ./fixtures/ember/*/node_modules",
    "fixtures:prepare": "node ./scripts/setup-fixtures.js ./fixtures/ember/app-template ./fixtures/ember/addon-template ./fixtures/ember/app-ts-template",
    "postinstall": "pnpm fixtures:prepare",
    "lint": "npm-run-all lint:*",
    "lint:tsc-src": "tsc --noEmit",
    "lint:tsc-test": "tsc --noEmit --project test/tsconfig.json",
    "test": "vitest --run --config ./vitest.config.ts --coverage",
    "test:slow": "vitest --run --config ./vitest.config.slow.ts",
    "version": "pnpm version"
  },
  "dependencies": {
    "find-up": "^6.3.0",
    "fixturify": "^3.0.0",
    "fixturify-project": "^5.2.0",
    "rimraf": "^4.4.1",
    "scenario-tester": "^2.1.2",
    "tmp": "^0.2.1"
  },
  "devDependencies": {
    "@vitest/coverage-c8": "^0.30.1",
    "ember-cli": "^4.11.0",
    "ember-data": "^4.11.3",
    "ember-source": "^4.12.0",
    "vitest": "^0.30.0"
  },
  "peerDependencies": {
    "typescript": "^5.0"
  },
  "packageManager": "pnpm@8.2.0",
  "volta": {
    "extends": "../../package.json"
  }
}
