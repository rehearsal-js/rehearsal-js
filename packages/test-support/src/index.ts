export * from './files.js';
export * from './library.js';
export * from './project.js';
export * from './scenarios.js';
