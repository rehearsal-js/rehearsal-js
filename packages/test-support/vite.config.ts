import { defineConfig } from 'vitest/config';

//eslint-disable-next-line import/no-default-export
export default defineConfig({
  test: {
    testTimeout: 100_000,
    hookTimeout: 100_000,
    watchExclude: ['package.json', '**/fixtures/**'],
    threads: false,
  },
});
