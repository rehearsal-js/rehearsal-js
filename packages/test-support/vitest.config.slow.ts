import { defineConfig } from 'vitest/config';


// explicit includes for slow test otherwise it'll run in `test`

//eslint-disable-next-line import/no-default-export
export default defineConfig({
  test: {
    testTimeout: 500_000,
    hookTimeout: 50_000,
    watchExclude: ['package.json', '**/fixtures/**'],
    include: ['./test/scenarios-app.test.ts']
  },
});
