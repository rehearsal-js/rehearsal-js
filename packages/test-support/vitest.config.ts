import { defineConfig } from 'vitest/config';

// we dont want to run these slow tests in this config
// these will all run by default in `test:slow` instead which is run in CI
const exclude = ['./test/scenarios-app.test.ts'];

//eslint-disable-next-line import/no-default-export
export default defineConfig({
  test: {
    testTimeout: 500_000,
    hookTimeout: 50_000,
    watchExclude: ['package.json', '**/fixtures/**'],
    sequence: {
      shuffle: true,
    },
    include: ['./test/**/*.test.ts'],
    exclude,
  },
});
