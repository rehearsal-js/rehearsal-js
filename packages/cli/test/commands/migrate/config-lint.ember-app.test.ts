import { resolve } from 'node:path';
import { readdirSync } from 'node:fs';
import { cosmiconfigSync } from 'cosmiconfig';

import { appScenarios, clean } from '@rehearsal/test-support';
import { Scenario, PreparedApp } from 'scenario-tester';
import { describe, beforeEach, afterEach, test, vi, expect } from 'vitest';
import {
  createCustomLintConfig,
  createLintConfig,
  extendLintConfig,
  skipConfigThatExtends,
  skipCustomConfigThatExtends,
} from '../../test-helpers/config-lint-test-utils.js';

describe('Task: config-lint -- ember-app-matrix', () => {
  appScenarios.forEachScenario((scenario: Scenario) => {
    describe(scenario.name, () => {
      let app: PreparedApp;
      let output = '';
      let explorerSync;

      beforeEach(async () => {
        app = await scenario.prepare();
        clean(app.dir);
        output = '';
        vi.spyOn(console, 'info').mockImplementation((chunk) => {
          output += `${chunk}\n`;
        });

        vi.spyOn(console, 'log').mockImplementation((chunk) => {
          output += `${chunk}\n`;
        });
      });

      afterEach(() => {
        vi.clearAllMocks();
        explorerSync = null;
        output = '';
      });

      test('create .eslintrc.js if not existed', async () => {
        await createLintConfig(app.dir);
        expect(readdirSync(app.dir)).toContain('.eslintrc.js');
        expect(readdirSync(app.dir)).toContain('.rehearsal-eslintrc.js');
        expect(output).matchSnapshot();
      });

      test('extends .eslintrc.js if existed', async () => {
        const oldConfig = `
        module.exports = {extends: []};
      `;
        await extendLintConfig(oldConfig, app.dir, '.eslintrc.js');

        expect(readdirSync(app.dir)).toContain('.eslintrc.js');
        expect(readdirSync(app.dir)).toContain('.rehearsal-eslintrc.js');
        // Do not use snapshot here since there is absolute path in output
        expect(output).toContain('extending Rehearsal default eslint-related config');

        /* eslint-disable-next-line @typescript-eslint/no-var-requires */
        const newConfig = require(resolve(app.dir, '.eslintrc.js')) as { extends: string[] };
        expect(newConfig.extends).toStrictEqual(['./.rehearsal-eslintrc.js']);
      });

      test('skip if .eslintrc.js exists with valid extends', async () => {
        const oldConfig = `
        module.exports = {extends: ["./.rehearsal-eslintrc.js"]};
      `;
        await skipConfigThatExtends(oldConfig, app.dir, '.eslintrc.js');
        expect(output).toContain('[SKIPPED] Create eslint config');
      });

      test('extends .eslintrc if existed', async () => {
        const oldConfig = `
        {"extends": []}
      `;
        await extendLintConfig(oldConfig, app.dir, '.eslintrc');

        expect(readdirSync(app.dir)).toContain('.eslintrc');
        expect(readdirSync(app.dir)).toContain('.rehearsal-eslintrc');
        // Do not use snapshot here since there is absolute path in output
        expect(output).toContain('extending Rehearsal default eslint-related config');

        explorerSync = cosmiconfigSync('');
        const loaded = explorerSync.load(resolve(app.dir, '.eslintrc'));
        const newConfig = loaded?.config as { extends: string[] };
        expect(newConfig.extends).toStrictEqual(['./.rehearsal-eslintrc']);
      });

      test('skip if .eslintrc exists with valid extends', async () => {
        const oldConfig = `
        {extends: ["./.rehearsal-eslintrc.js"]}
      `;
        await skipConfigThatExtends(oldConfig, app.dir, '.eslintrc');

        expect(output).toContain('[SKIPPED] Create eslint config');
      });

      test('extends .eslintrc.json if existed', async () => {
        const oldConfig = `
        {"extends": []}
      `;
        await extendLintConfig(oldConfig, app.dir, '.eslintrc.json');

        expect(readdirSync(app.dir)).toContain('.eslintrc.json');
        expect(readdirSync(app.dir)).toContain('.rehearsal-eslintrc.json');
        // Do not use snapshot here since there is absolute path in output
        expect(output).toContain('extending Rehearsal default eslint-related config');

        explorerSync = cosmiconfigSync('');
        const loaded = explorerSync.load(resolve(app.dir, '.eslintrc.json'));
        const newConfig = loaded?.config as { extends: string[] };
        expect(newConfig.extends).toStrictEqual(['./.rehearsal-eslintrc.json']);
      });

      test('skip if .eslintrc.json exists with valid extends', async () => {
        const oldConfig = `
        {"extends": ["./.rehearsal-eslintrc.js"]}
      `;
        await skipConfigThatExtends(oldConfig, app.dir, '.eslintrc.json');

        expect(output).toContain('[SKIPPED] Create eslint config');
      });

      test('extends .eslintrc.yml if existed', async () => {
        const oldConfig = `
        extends: []
      `;
        await extendLintConfig(oldConfig, app.dir, '.eslintrc.yml');

        expect(readdirSync(app.dir)).toContain('.eslintrc.yml');
        expect(readdirSync(app.dir)).toContain('.rehearsal-eslintrc.yml');
        // Do not use snapshot here since there is absolute path in output
        expect(output).toContain('extending Rehearsal default eslint-related config');

        explorerSync = cosmiconfigSync('');
        const loaded = explorerSync.load(resolve(app.dir, '.eslintrc.yml'));
        const config = loaded?.config as { extends: string[] };

        /* eslint-disable-next-line @typescript-eslint/no-var-requires */
        expect(config.extends).toStrictEqual(['./.rehearsal-eslintrc.yml']);
      });

      test('skip if .eslintrc.yml exists with valid extends', async () => {
        const oldConfig = `
        extends: ["./.rehearsal-eslintrc.js"]
      `;
        await skipConfigThatExtends(oldConfig, app.dir, '.eslintrc.yml');
        expect(output).toContain('[SKIPPED] Create eslint config');
      });

      test('extends .eslintrc.yaml if existed', async () => {
        const oldConfig = `
        extends: []
      `;
        await extendLintConfig(oldConfig, app.dir, '.eslintrc.yaml');

        expect(readdirSync(app.dir)).toContain('.eslintrc.yaml');
        expect(readdirSync(app.dir)).toContain('.rehearsal-eslintrc.yaml');
        // Do not use snapshot here since there is absolute path in output
        expect(output).toContain('extending Rehearsal default eslint-related config');

        explorerSync = cosmiconfigSync('');
        const loaded = explorerSync.load(resolve(app.dir, '.eslintrc.yaml'));
        const config = loaded?.config as { extends: string[] };

        /* eslint-disable-next-line @typescript-eslint/no-var-requires */
        expect(config.extends).toStrictEqual(['./.rehearsal-eslintrc.yaml']);
      });

      test('skip if .eslintrc.yaml exists with valid extends', async () => {
        const oldConfig = `
        extends: ["./.rehearsal-eslintrc.js"]
      `;
        await skipConfigThatExtends(oldConfig, app.dir, '.eslintrc.yaml');

        expect(output).toContain('[SKIPPED] Create eslint config');
      });

      test('run custom config command with user config provided', async () => {
        const customConfig = {
          migrate: {
            setup: {
              lint: { command: 'touch', args: ['custom-lint-config-script'] },
            },
          },
        };
        await createCustomLintConfig(app.dir, customConfig);

        expect(readdirSync(app.dir)).toContain('custom-lint-config-script');
        expect(output).toMatchSnapshot();
      });

      test('postLintSetup hook from user config', async () => {
        const customConfig = {
          migrate: {
            setup: {
              lint: { command: 'touch', args: ['custom-lint-config-script'] },
              postLintSetup: { command: 'mv', args: ['custom-lint-config-script', 'foo'] },
            },
          },
        };

        await createCustomLintConfig(app.dir, customConfig);

        expect(readdirSync(app.dir)).toContain('foo');
        expect(readdirSync(app.dir)).not.toContain('custom-lint-config-script');
        expect(output).toMatchSnapshot();
      });

      test('skip if custom config and .eslintrc.js exist', async () => {
        const customConfig = {
          migrate: {
            setup: {
              lint: { command: 'touch', args: ['custom-lint-config-script'] },
            },
          },
        };
        await skipCustomConfigThatExtends(app.dir, customConfig);

        expect(readdirSync(app.dir)).not.toContain('custom-lint-config-script');
        expect(output).toContain('[SKIPPED] Create eslint config');
      });
    });
  });
});
