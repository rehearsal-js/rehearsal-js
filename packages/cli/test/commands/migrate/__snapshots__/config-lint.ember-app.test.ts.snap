// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Task: config-lint -- ember-app-matrix > app > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > app > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > app > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > appEmber4 > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > appEmber4 > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > appEmber4 > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > appWithInRepoAddon > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > appWithInRepoAddon > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > appWithInRepoAddon > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > appWithInRepoEngine > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > appWithInRepoEngine > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > appWithInRepoEngine > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > ember3-app > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > ember3-app > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > ember3-app > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > ember3-appWithInRepoAddon > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > ember3-appWithInRepoAddon > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > ember3-appWithInRepoAddon > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > ember3-appWithInRepoEngine > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > ember3-appWithInRepoEngine > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint -- ember-app-matrix > ember3-appWithInRepoEngine > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > ember4App > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > ember4App > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > ember4App > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > emberApp > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > emberApp > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > emberApp > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > emberAppWithInRepoAddon > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > emberAppWithInRepoAddon > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > emberAppWithInRepoAddon > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > emberAppwithInRepoEngine > create .eslintrc.js if not existed 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js, extending Rehearsal default eslint-related config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > emberAppwithInRepoEngine > postLintSetup hook from user config 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[DATA] Run postLintSetup from config
[SUCCESS] Create eslint config
"
`;

exports[`Task: config-lint for ember app > emberAppwithInRepoEngine > run custom config command with user config provided 1`] = `
"[STARTED] Install dependencies
[DATA] Installing devDependencies: @types/node, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, eslint, eslint-config-prettier, eslint-plugin-prettier, prettier, typescript
[SUCCESS] Install dependencies
[STARTED] Create eslint config
[DATA] Create .eslintrc.js from config
[SUCCESS] Create eslint config
"
`;
